CIP-0144/README.md

```diff
@@ -0,0 +1,356 @@
+In this CIP we want to be able to define an API without committing to a specific transport layer. Implementors of this API can choose to support this API through several transports such as: HTTP, an injected Javascript object, JSON-RPC etc.
+Furthermore, we want to use JSON-schema to clearly define the types that each method, or operation, expects to receive or returns.
+To keep the specification abstract we will use the word "operation" to describe the actions supported by the API: these would map to "endpoints" for an HTTP implementation of the API, or to "methods" for an implementation based on an injected Javascript object.
+
+
+We will use the following schema to define operations:
+
+```json
+{
+  "operation": { "type": "string" },
+  "request": { "type": "object" },
+  "response": { "type": "object" },
+  "errors": {
+    "type": "array",
+    "items": { "type": "object" }
+  }
+}
+```
+
+where `operation` will be the identifier for the operation, `request` and `response` will be the JSON-schemas for the request and response types respectively, and `errors` will be a list of schemas for the errors that the operation may raise.
+We do not add an explicit scope to operation names, we do however encourage transport-specific implementations of this API to pick the scoping structure that makes more sense to them.
+
+We will reference several JSON schemas throughout the document, these are:
+
+- [CIP-116 | Standard JSON encoding for Domain Types](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0116) which provides a JSON encoding of Cardano ledger types. Note that this CIP defines a schema for each ledger era. When referring to a type from this schema we refer to an `anyOf` of all the schemas in which that type is defined.
+- [appendix](#appendix) in which we define schemas for types required by the connection API and the error types
+
+We will use an identifier in the anchor to refer to the schema where each type is defined. For example, if we want to reference the `Transaction` type, as defined in CIP-116 we will use the following schema reference `{ "$ref": "#/cip-116/Transaction" }`.
+
+If an operation does not require any arguments as part of it's request, or does not return a meaningful response, we will represent this as an `{}` on the respective operation field. The arguments to an operation will always be represented as an object. If a field is not marked as `required`, then that argument is to be considered optional.
+
+For each operation we will provide some details on how the implementation should behave. Note that some of these are taken verbatim from CIP-30.
+
+### Connection API
+
+The role of the connection API is to provide generic information about the wallet and to allow the user to opt into the functionalities (or extensions) that they want the wallet to provide.
+
+##### Enable
+
+```
+{
+  "operation": "enable",
+  "request": { "$ref": "#/appendix/Extensions" },
```

Comment:
just a thought, would there be a usecase where a dApp doesnt request a whole extension?
perhaps a dApp only wants a couple methods, maybe this would just add unneeded complexity

Reply:
Discussed in meeting
we can consider this in the future, or when we have more feedback from stakeholders